Mage.Sets/src/mage/cards/m/MirkoObsessiveTheorist.java

```diff
@@ -0,0 +1,85 @@
+package mage.cards.m;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfYourEndStepTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.common.SurveilTriggeredAbility;
+import mage.abilities.effects.common.InfoEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToBattlefieldWithCounterTargetEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.common.FilterCreatureCard;
+import mage.filter.predicate.mageobject.PowerPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.target.targetadjustment.TargetAdjuster;
+
+import java.util.UUID;
+
+/**
+ * @author PurpleCrowbar
+ */
+public final class MirkoObsessiveTheorist extends CardImpl {
+
+    public MirkoObsessiveTheorist(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{U}{B}");
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.VAMPIRE, SubType.DETECTIVE);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Whenever you surveil, put a +1/+1 counter on Mirko, Obsessive Theorist.
+        this.addAbility(new SurveilTriggeredAbility(new AddCountersSourceEffect(CounterType.P1P1.createInstance())));
+
+        // At the beginning of your end step, you may return target creature card with power less than Mirko's from your graveyard to the battlefield with a finality counter on it.
+        Ability ability = new BeginningOfYourEndStepTriggeredAbility(
+                new ReturnFromGraveyardToBattlefieldWithCounterTargetEffect(CounterType.FINALITY.createInstance()),true
+        );
+        ability.setTargetAdjuster(MirkoObsessiveTheoristAdjuster.instance);
```

Comment:
Look at GracefulRestoration - it uses target filter without adjuster. Why not use same way?

Reply:
Maybe like EarthshakerKhenraPredicate

Reply:
Using a custom predicate and setText as in the case with Earthshaker Khenra:
Rules text:
![image](https://github.com/magefree/mage/assets/26198472/c360fa8f-9d04-4d95-bce8-47c8b094da6a)
Target text:
![image](https://github.com/magefree/mage/assets/26198472/28529608-d397-4af2-87db-8f05d7049ea4)
Confirming resolution:
![image](https://github.com/magefree/mage/assets/26198472/efdf539f-710c-4009-b8ac-2346a7caac4a)

The resolution confirming is still wordy but presumably consistent with other cards so not important